subscribe.go

```diff
@@ -54,17 +56,65 @@ func (f HandlerFunc) Handle(msg SubMessage) error {
 
 // ----------------------------------------------------------------------------
 
+// SubStart defines starting position to consume messages.
+type SubStart int
+
+// SubStart options.
+const (
+	Newest SubStart = iota
+	Oldest
+)
+
+// SubOptions holds subscription options.
+type SubOptions struct {
+	// Start defines starting position to consume messages.
+	// Default: SubscriptionNewest
+	Start SubStart
+}
+
+// NewSubOptions builds SubOptions.
+//
+// It is meant to be used by pubsub implementations:
+//
+//   func (s *SubImpl) Subscribe(..., options ...bps.SubOption) error {
+//     opts := bps.NewSubOptions(options)
+//     ...
+//   }
+//
+func NewSubOptions(opts []SubOption) *SubOptions {
+	o := &SubOptions{
+		Start: Newest,
+	}
+	for _, opt := range opts {
+		opt(o)
+	}
+	return o
+}
+
+// SubOption defines a single subscription option.
+type SubOption func(*SubOptions)
+
+// Start configures subscription start position.
+func Start(start SubStart) SubOption {
```

Comment:
This is a neat approach for some libs but only of the underlying struct is private, I am not too fussed and OK to leave it but since it adds a lot (!) of extra funcs, I would prefer less colliding names, i.e. `StartAt(..)` vs `Start(..)`

Reply:
Already thought about StartAt, will rename.

Struct is kept public only to share between implementations.

Could be smth like `internal/options.SubOption/SubOptions`, but that's a bit too much, so kept it in root pkg.